xclim/indices/helpers.py

```diff
@@ -550,3 +552,82 @@ def _gather_lon(da: xr.DataArray) -> xr.DataArray:
             "Try passing it explicitly (`lon=ds.lon`)."
         )
         raise ValueError(msg) from err
+
+
+def resample_map(
+    obj: xr.DataArray | xr.Dataset,
+    dim: str,
+    freq: str,
+    func: Callable,
+    map_blocks: bool | Literal["from_context"] = "from_context",
+    resample_kwargs: dict | None = None,
+    map_kwargs: dict | None = None,
+) -> xr.DataArray | xr.Dataset:
+    r"""
+    Wraps xarray's resample(...).map() with a :py:func:`xarray.map_blocks`, ensuring the chunking is appropriate using flox.
+
+    Parameters
+    ----------
+    obj: DataArray or Dataset
+        The xarray object to resample.
+    dim: str
+        Dimension over which to resample.
+    freq : str
+        Resampling frequency along `dim`.
+    func : callable
+        Function to map on each resampled group.
+    map_blocks: bool or "from_context"
+        If True, the resample().map() call is wrapped inside a `map_blocks`.
+        If False, this does not do anything special.
+        If "from_context", xclim's "resample_map_blocks" option is used.
+        If the object is not using dask, this is set to False.
+    resample_kwargs: dict, optional
+        Other arguments to pass to `obj.resample()`.
+    map_kwargs: dict, optional
+        Arguments to pass to `map`.
+
+    Returns
+    -------
+    Resampled object.
+    """
+    resample_kwargs = resample_kwargs or {}
+    map_kwargs = map_kwargs or {}
+    if map_blocks == "from_context":
+        map_blocks = OPTIONS[MAP_BLOCKS]
+
+    if not uses_dask(obj) or not map_blocks:
+        return obj.resample({dim: freq}, **resample_kwargs).map(func, **map_kwargs)
+
+    try:
+        from flox.xarray import rechunk_for_blockwise
+    except ImportError as err:
+        raise ValueError(f"Using {MAP_BLOCKS}=True requires flox.") from err
```

Comment:
Linters complain about f-strings in logging/exception calls.

```suggestion
        msg = f"Using {MAP_BLOCKS}=True requires flox."
        raise ValueError(msg) from err
```